### PR TITLE
Refine SchemaManagement Project 

### PR DESCRIPTION
**Pull request by QuanWanxx**
1. Add tests.
2. Handle more corner cases, including null, empty or invalid inputs, or invalid schema content.

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/DataConverter/DefaultSchemaConverter.cs [hunk 1]
```diff
@@ -125,9 +126,9 @@ private JArray ProcessArrayObject(JToken arrayItem, FhirParquetSchemaNode schema
         {
             if (arrayItem is not JArray fhirArrayObject)
             {
-                _diagnosticLogger.LogError($"Current FHIR object is not a valid JArray: {schemaNode.GetNodePath()}.");
-                _logger.LogInformation($"Current FHIR object is not a valid JArray: {schemaNode.GetNodePath()}.");
-                throw new ParquetDataProcessorException($"Current FHIR object is not a valid JArray: {schemaNode.GetNodePath()}.");
+                _diagnosticLogger.LogError($"Current FHIR object is not a valid JArray: {arrayItem.Path}.");
+                _logger.LogInformation($"Current FHIR object is not a valid JArray: {arrayItem.Path}.");
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/DataConverter/DefaultSchemaConverter.cs:130]: When we use `item.Path` and when we use `item` directly in logging message?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaValidator.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Health.Fhir.Synapse.SchemaManagement.Parquet
+{
+    public class FhirParquetSchemaValidator
+    {
+        public static bool Validate(string schemaKey, FhirParquetSchemaNode rootNode, ref string error)
+        {
+            if (schemaKey == null)
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaValidator.cs:15]: Maybe IsNullOrEmpty?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaValidator.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            {
+                error = "The schema key cannot be null.";
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaValidator.cs:17]: What about wrapping the error message and bool value into a  ValidationResult { bool Success; string Message }?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/DataConverter/CustomSchemaConverter.cs [hunk 4]
```diff
@@ -45,8 +45,8 @@ public class CustomSchemaConverter : IDataSchemaConverter
         {
             EnsureArg.IsNotNull(schemaConfiguration, nameof(schemaConfiguration));
 
-            _diagnosticLogger = EnsureArg.IsNotNull(diagnosticLogger, nameof(diagnosticLogger));
             _logger = EnsureArg.IsNotNull(logger, nameof(logger));
```
Comment (hongyh13) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/DataConverter/CustomSchemaConverter.cs:48]: Is it possile that `JosnProcessor()` throw an exception? do we need lazy initialization for it?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaValidator.cs (hunk 3 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaValidator.cs]: Sure, updated.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaValidator.cs (hunk 2 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/Parquet/FhirParquetSchemaValidator.cs]: Fixed, thanks.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/DataConverter/DefaultSchemaConverter.cs (hunk 1 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataProcessor/DataConverter/DefaultSchemaConverter.cs:140]: That actually is a bug, fixed, thanks.

**Review by moria97 — APPROVED** [review]
